lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1390,6 +1392,106 @@ extension DomCSSRuleListExtension on DomCSSRuleList {
   external int get length;
 }
 
+/// Trusted Types API
+
+/*
+// Trusted Types API (TrustedTypePolicy, TrustedScript, TrustedScriptURL)
+// https://developer.mozilla.org/en-US/docs/Web/API/TrustedTypesAPI
+*/
+
+/// A factory to create `TrustedTypePolicy` objects.
+@JS()
+@staticInterop
+abstract class DomTrustedTypePolicyFactory {}
+
+/// (Some) methods of the [DomTrustedTypePolicyFactory]:
+extension DomTrustedTypePolicyFactoryExtension on DomTrustedTypePolicyFactory {
+  /// createPolicy
+  external DomTrustedTypePolicy createPolicy(
+    String policyName,
+    DomTrustedTypePolicyOptions? policyOptions,
+  );
+}
+
+/// Options to create a trusted type policy.
+@JS()
+@staticInterop
+@anonymous
+abstract class DomTrustedTypePolicyOptions {
+  /// Constructs a TrustedPolicyOptions object in JavaScript.
+  ///
+  /// The following properties need to be manually wrapped in [allowInterop]
+  /// before being passed to this constructor: [createScriptURL].
+  external factory DomTrustedTypePolicyOptions({
+    DomCreateScriptUrlOptionFn? createScriptURL,
+  });
+}
+
+/// Type of the function to configure createScriptURL
+typedef DomCreateScriptUrlOptionFn = String? Function(String input);
+
+/// An instance of a TrustedTypePolicy
```

Comment:
nit: "instance" typically means the object created at runtime. This is more a "type binding".

Reply:
Do you think describing JS-interop classes as "type bindings" helps with the readability of the documentation? It might be more technically accurate (for Dart language quirks), but I'm not sure it helps actual users of the API much.

When a developer ends up with a "DomTrustedTypePolicy" object, they'll be handling "an instance" of a TrustedTypePolicy, which just happened to be created in JS rather than in Dart, and not the type binding (which is the language mechanism to eventually get that "Object" into their code).

`dart:html` never mentions the techniques it uses to bind its code to the browser APIs (other than mentioning the `@Native` annotation)

I don't mind too much, (and nothing else in this file is documented), so I went by something in between:

```
/// Dart binding of a JavaScript TrustedTypePolicy instance.
```

Reply:
Hmm, good point. "type binding" is also not the best choice here. The class is still not an instance, because instance means object or value. I'd just extract the docs from MDN, something like: "Defines a group of functions that create TrustedType objects."

Reply:
Good idea, ended up copying the description from the MDN (I think all our JS-interop docs should literally be automatic links to the correct MDN/API doc page :P)